scripts/objectron_eval.py

```diff
@@ -0,0 +1,228 @@
+                    num_matched += 1
+                    pixel_error = self.evaluate_2d(box_point_2d, instances[index])
+                    # If you only compute the 3D bounding boxes from RGB images,
+                    # your 3D keypoints may be upto scale. However the ground truth
+                    # is at metric scale. There is a hack to re-scale your box using
+                    # the ground planes (assuming your box is sitting on the ground).
+                    # However many models learn to predict depths and scale correctly.
+                    scale = self.compute_scale(box_point_3d, plane)
+                    box_point_3d = box_point_3d * scale
+                    azimuth_error, polar_error, iou, add, adds = self.evaluate_3d(
+                        box_point_3d, instances_3d[index])
+                else:
+                    pixel_error = _MAX_PIXEL_ERROR
+                    azimuth_error = _MAX_AZIMUTH_ERROR
+                    polar_error = _MAX_POLAR_ERROR
+                    iou = 0.
+                    add = _MAX_DISTANCE
+                    adds = _MAX_DISTANCE
+
+                iou_hit_miss.record_hit_miss(iou)
+                add_hit_miss.record_hit_miss(add, greater=False)
+                adds_hit_miss.record_hit_miss(adds, greater=False)
+                pixel_hit_miss.record_hit_miss(pixel_error, greater=False)
+                azimuth_hit_miss.record_hit_miss(azimuth_error, greater=False)
+                polar_hit_miss.record_hit_miss(polar_error, greater=False)
+
+            self._iou_ap.append(iou_hit_miss, len(instances))
+            self._pixel_ap.append(pixel_hit_miss, len(instances))
+            self._azimuth_ap.append(azimuth_hit_miss, len(instances))
+            self._polar_ap.append(polar_hit_miss, len(instances))
+            self._add_ap.append(add_hit_miss, len(instances))
+            self._adds_ap.append(adds_hit_miss, len(instances))
+            self._matched += num_matched
+
+
+def main(argv):
+    if len(argv) > 1:
+        raise app.UsageError('Too many command-line arguments.')
+
+    class_name = FLAGS.eval_data.split(os.path.sep)[-2]
+    correct_class_index = OBJECTRON_CLASSES.index(class_name)
+
+    ie = IECore()
+    object_detector = Detector(ie, FLAGS.detection_model, 0.5)
+    regression_model = Regressor(ie, FLAGS.regression_model)
+
+    print('Starting evaluation')
+    evaluator = Torchdet3dEvaluator(
+        object_detector, regression_model, correct_class_index)
+
+    ds = tf.data.TFRecordDataset(
+        glob.glob(FLAGS.eval_data)).take(FLAGS.max_num)
+    batch = []
+    for serialized in tqdm.tqdm(ds):
+        batch.append(serialized.numpy())
+        if len(batch) == FLAGS.batch_size:
+            evaluator.evaluate(batch)
+            batch.clear()
+
+    if batch:
+        evaluator.evaluate(batch)
+
+    evaluator.finalize()
+    evaluator.write_report()
+
+
+if __name__ == '__main__':
+    flags.mark_flag_as_required('report_file')
```

Comment:
and change this to arparse representation to align all other scripts if it's not a hassle